packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/class_details.dart

```diff
@@ -119,34 +122,49 @@ class _RetainedSizeColumn extends ColumnData<StatsByPathEntry> {
       )!;
 }
 
+class _RetainingPathTableColumns {
+  _RetainingPathTableColumns(this.isDiff);
+
+  final bool isDiff;
+  late final shallowSizeColumn = _ShallowSizeColumn(isDiff);
+  late final columnList = <ColumnData<StatsByPathEntry>>[
+    _RetainingPathColumn(),
+    _InstanceColumn(isDiff),
+    shallowSizeColumn,
+    _RetainedSizeColumn(isDiff),
+  ];
+}
+
 class _RetainingPathTable extends StatelessWidget {
   const _RetainingPathTable({
     Key? key,
     required this.entries,
     required this.selection,
+    required this.isDiff,
   }) : super(key: key);
 
   final List<StatsByPathEntry> entries;
   final ValueNotifier<StatsByPathEntry?> selection;
+  final bool isDiff;
 
-  static final _shallowSizeColumn = _ShallowSizeColumn();
-  static late final List<ColumnData<StatsByPathEntry>> _columns =
-      <ColumnData<StatsByPathEntry>>[
-    _RetainingPathColumn(),
-    _InstanceColumn(),
-    _shallowSizeColumn,
-    _RetainedSizeColumn(),
-  ];
+  static final _columns = <String, _RetainingPathTableColumns>{};
+  static _RetainingPathTableColumns _obtainColumns(
```

Comment:
naming nit: r/obtain/generate

Reply:
I agree 'obtain' is not the best name, however 'generate' sounds confusing to me as it implies the method will always create new columns, while it is not true. Renamed to '_columns'.